### PR TITLE
Variable comm radius experiment

### PR DESCRIPTION
**Pull request by MILeach**
Introduces an extra experiment testing the effect of varying comm radius with a fixed environment size and population size.

Update fgpu2 repo version once agent sorting brought in

a, b, d

3 runs per data point

Run on bessemer

Tidy up repo

Separate folder for each GPU

Commit current master to GTC branch

Vis screenshots of initial & equilibrium state

### CODE REVIEW

File: src/main.cu [hunk 1]
```diff
@@ -292,11 +292,65 @@ bool experiment_density_spatial(const custom_cli cli) {
...
+
+    return success;
+}
+
 
 int main(int argc, const char ** argv) {
     // Custom arg parsing, to prevent the current F2 arg parsing from occuring. 
     // @todo - improve arg parsing within F2. 
     custom_cli cli = parse_custom_cli(argc, argv);
+    cli.steps = 200;
```
Comment (ptheywood) [inline src/main.cu]: This needs to change the defualts instead.

File: src/main.cu [hunk 2]
```diff
@@ -292,11 +292,65 @@ bool experiment_density_spatial(const custom_cli cli) {
...
+    // const std::vector<float> ENV_WIDTHS = {8.f, 12.f, 16.f, 20.f};
+    // const std::vector<float> ENV_WIDTHS = {8.f, 12.f, 16.f, 20.f, 30.f, 40.f, 50.f, 60.f, 70.f, 80.f, 90.f, 100.f};
+    const float ENV_WIDTH = 40.0f;  
+
+
+    const std::vector<float> comm_radii = {2.0f, 4.0f, 6.0f, 8.0f, 10.0f, 12.0f, 14.0f, 16.0f, 18.0f, 20.0f};
+
+    // Select the models to execute.
+    std::map<std::string, std::function<void(const RunSimulationInputs, RunSimulationOutputs&)>> MODELS = {
+        //{std::string("circles_spatial3D"), run_circles_spatial3D},
```
Comment (Robadob) [inline src/main.cu]: Why are these commented out? Should it not be configurable somehow?

**Comment by Robadob** [comment]
At the bottom of main.cu, there's a todo list. Can that be moved to issues on the repo?

**Review by Robadob — APPROVED** [review]
I see no other clear problems, than those already noted by Pete and myself. But i'm not familiar with these benchmark repos in the slightest.

I will try to run it on Mavericks, as Paul requested in a bit.

**Edit:** It's now running on Mav, guess I'll check back tomorrow, it's slowed down since it got to ~140/228 on the first experiment.

**Edit2:** Didn't run it inside Tmux, so the run died when I turned off my pc last night.

**Comment by Robadob** [comment]
In readme there is a typo 'Generte'

**Comment by Robadob** [comment]
image
A todo in the console output?

**Comment by ptheywood** [comment]
The todos were me tracking things as i was quickly implementing this pre GTC and they just never got tided up. Some should probably promoted to issue(s) while some can probably just be dismissed. The printf'd todo can at the very least be disabled.

**Pull request merged**